src/sphinx_substitution_extensions/__init__.py

```diff
@@ -27,15 +30,16 @@
 # See https://github.com/PyCQA/doc8/pull/34.
 _SUBSTITUTION_OPTION_NAME = 'substitutions'
 
-if 'prompt' not in _EXISTING_DIRECTIVES:
-    MESSAGE = (
-        'sphinx-prompt must be in the conf.py extensions list before '
-        'sphinx_substitution_extensions'
-    )
-    LOGGER.error(MESSAGE)
-
-_EXISTING_PROMPT_DIRECTIVE: Directive = _EXISTING_DIRECTIVES['prompt']
-
+_EXISTING_PROMPT_DIRECTIVE = Directive
```

Comment:
Why have this? Is it so that we do not error when subclassing `_EXISTING_PROMPT_DIRECTIVE` for `SubstitutionPrompt`?

If so, I wonder, can we do better?
In particular, it is a bit confusing that we have a variable `_EXISTING_PROMPT_DIRECTIVE`, which is very much not an existing prompt directive.

I also expect that `mypy` will have problems with this but I cannot test that now as it appears that PyPI is having issues.

For example:

* Adding a comment explaining this line (bare minimum), or preferably
* Only define `SubstitutionCodeBlock` if we have `sphinx-prompt`

Reply:
To not add the comment, I've done the following:

- Moved ``SubstitutionPrompt`` into a separate file ``extras.py``.
- Load the directive conditionally, only if ``sphinx-prompt`` is installed.
- Created ``shared.py`` for variables and functions that are common for both directives.

What do you think?